src/WebJobs.Script/Config/WorkerConcurrencyOptions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class WorkerConcurrencyOptions : Hosting.IOptionsFormatter
+    {
+        public WorkerConcurrencyOptions()
+        {
+            // Setting deafault values
+            LatencyThreshold = TimeSpan.FromSeconds(1);
+            AdjustmentPeriod = TimeSpan.FromSeconds(10);
+            CheckInterval = TimeSpan.FromSeconds(1);
+            HistorySize = 10;
+            HistoryThreshold = 1F;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether worker concurrency
+        /// is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the latency threshold dictating when worker channel is overloaded.
+        /// </summary>
+        public TimeSpan LatencyThreshold { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time when worker
+        /// channels will start to monitor again after adding a new worker.
+        /// </summary>
+        public TimeSpan AdjustmentPeriod { get; set; }
+
+        /// <summary>
+        /// Gets or sets interval to check worker channel state.
+        /// </summary>
+        public TimeSpan CheckInterval { get; set; }
+
+        /// <summary>
+        /// Gets or sets the history size to store workers channel states.
+        /// </summary>
+        public int HistorySize { get; set; }
+
+        /// <summary>
+        /// Gets or sets the history threshold.
+        /// E. g. value equal to 1.0 means all states in history should be
+        /// overloaded to consider worker state as overloaded.
+        /// </summary>
+        public float HistoryThreshold { get; set; }
```

Comment:
Trying to come up with a more descriptive name for this. Perhaps NewWorkerThreshold? I'd describe in this xml doc as "Gets or sets the threshold dictating when a new worker will be added. Value should be between 0 and 1 indicating the percentage of overloaded channel latency samples required to trigger a addition of a new worker".

Reply:
Also should add a range check on this